src/workers/cdc/primary.ts

```diff
@@ -317,6 +317,51 @@ const handleCommentUpvoteChange = async (
   }
 };
 
+const handleCommentDownvoteChange = async (
+  con: DataSource,
+  logger: FastifyBaseLogger,
+  data: ChangeMessage<UserComment>,
+): Promise<void> => {
+  switch (data.payload.op) {
+    case 'c':
+      await triggerTypedEvent(logger, 'api.v1.comment-downvoted', {
+        commentId: data.payload.after.commentId,
+        userId: data.payload.after.userId,
+      });
+      break;
+    case 'u': {
+      const isDownvoteCanceled = data.payload.after.vote === UserVote.None;
+
+      if (isDownvoteCanceled) {
+        await triggerTypedEvent(logger, 'api.v1.comment-downvote-canceled', {
+          commentId: data.payload.before.commentId,
+          userId: data.payload.before.userId,
+        });
+      } else {
+        await triggerTypedEvent(logger, 'api.v1.comment-downvoted', {
+          commentId: data.payload.after.commentId,
+          userId: data.payload.after.userId,
+        });
+      }
+      break;
+    }
+    case 'd': {
+      const wasDownvoted = data.payload.before.vote === UserVote.Down;
+
+      if (wasDownvoted) {
+        await triggerTypedEvent(logger, 'api.v1.comment-downvote-canceled', {
```

Comment:
Just checking but with the typed ones did we not say we always send the entity as a whole to make it more streamlined for future additions? 
(can't fully recall if that was the decision)

Reply:
I think at the end it was more of a make it constant inside the subscriptions of the same topic. The [current typed ones ](https://github.com/dailydotdev/daily-api/blob/0c623e2911702484d5a3b0887071c7f177170142/src/common/typedPubsub.ts#L10)are also defined like that.